pkg/cli/cmd/taskrun/list.go

```diff
@@ -0,0 +1,267 @@
+    tkn-results taskrun list --limit=20
+
+List TaskRuns for a specific task:
+    tkn-results taskrun list foo -n namespace
+
+List TaskRuns with partial task name match:
+    tkn-results taskrun list build -n namespace
+
+List TaskRuns for a specific PipelineRun:
+    tkn-results taskrun list --pipelinerun my-pipeline-run -n namespace
+`
+	cmd := &cobra.Command{
+		Use:     "list [task-name]",
+		Aliases: []string{"ls"},
+		Short:   "List TaskRuns in a namespace",
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Example: eg,
+		PreRunE: func(_ *cobra.Command, args []string) error {
+			c, err := config.NewConfig(p)
+			if err != nil {
+				return err
+			}
+			opts.Client, err = client.NewRESTClient(c.Get())
+			if err != nil {
+				return err
+			}
+
+			if opts.Limit < 5 || opts.Limit > 1000 {
+				return errors.New("limit should be between 5 and 1000")
+			}
+
+			// Validate label format if provided
+			if opts.Label != "" {
+				return common.ValidateLabels(opts.Label)
+			}
+
+			if len(args) > 0 {
+				opts.ResourceName = args[0]
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return listTaskRuns(cmd.Context(), p, opts)
+		},
+	}
+
+	cmd.Flags().Int32VarP(&opts.Limit, "limit", "", 50, "Maximum number of TaskRuns to return (must be between 5 and 1000, default is 50)")
+	cmd.Flags().BoolVarP(&opts.AllNamespaces, "all-namespaces", "A", false, "List TaskRuns from all namespaces")
+	cmd.Flags().StringVarP(&opts.Label, "label", "L", "", "Filter by label (format: key=value[,key=value...])")
+	cmd.Flags().StringVarP(&opts.PipelineRun, "pipelinerun", "", "", "Filter TaskRuns by PipelineRun name. Note that multiple PipelineRuns can have the same name, so this will return TaskRuns from all PipelineRuns with the matching name.")
+	cmd.Flags().BoolVar(&opts.SinglePage, "single-page", true, "Return only a single page of results")
+
+	return cmd
+}
+
+func listTaskRuns(ctx context.Context, p common.Params, opts *options.ListOptions) error {
+	// Build filter string
+	filter := common.BuildFilterString(opts, "taskrun")
+
+	// Handle all namespaces
+	parent := fmt.Sprintf("%s/results/-", p.Namespace())
+	if opts.AllNamespaces {
+		parent = "*/results/-"
+	}
+
+	// Create initial request
+	req := &pb.ListRecordsRequest{
+		Parent:   parent,
+		Filter:   filter,
+		OrderBy:  "create_time desc",
+		PageSize: opts.Limit,
+	}
+
+	// Create record client
+	recordClient := records.NewClient(opts.Client)
+
+	// If single page is requested, just fetch and print
+	if opts.SinglePage {
+		resp, err := recordClient.ListRecords(ctx, req)
+		if err != nil {
+			return err
+		}
+
+		stream := &cli.Stream{
+			Out: os.Stdout,
+			Err: os.Stderr,
+		}
+
+		// Parse records to TaskRuns before printing
+		trs, err := parseRecordsToTr(resp.Records)
+		if err != nil {
+			return err
+		}
+		return printFormattedTr(stream, trs, clockwork.NewRealClock(), opts.AllNamespaces, false)
```

Comment:
Can't we run that from the for loop below and break after the first iteration if opts.SinglePage is true, instead of repeating it? The check can be added to line 191.

Reply:
Sure, I will do it. Thanks!